### PR TITLE
fix: Issue in the tickets fragment when the internet is turned off.

### LINKED ISSUE #2801: Issue in the Tickets Fragment when there is no internet.
**Describe the bug**
When the internet is turned off ,  The tickets fragment seems to still show the shimmer not the No-Internet card .
**To Reproduce**
1. Turn the internet off.
2. Open the EventYay app.
3. Click on the Tickets icon on the Bottom Navigation bar. 
4. You can observe that the shimmer still keeps loading and the No-Internet card is never shown.
**Expected behavior**
When the internet is turned off , the tickets fragment should display the No-Internet card.

**Screenshots**

**Smartphone Info:**
issue

|Device         Samsung J7 duo
|Android Version|< Oreo 8.1   |

**Additional context**

**Would you like to work on the issue?**
 Yes

### PR DESCRIPTION
**Pull request by VamsiKrishnaCommits**
Fixes #2801 

Changes: 
The tickets fragment now displays as expected i.e. displays no internet card not the shimmer loading when the internet is turned off.

Screenshots for the change:
improved

### CODE REVIEW

**Comment by VamsiKrishnaCommits** [comment]
@mariobehling @iamareebjamal @liveHarshit please review this PR

**Comment by iamareebjamal** [comment]
The build is failing. How can you expect us to review the PR?

**Comment by VamsiKrishnaCommits** [comment]
> The build is failing. How can you expect us to review the PR?

sir, travis build is failing just for adding if and else statement. I tried to make the minimum changes . While I can run the code with no errors seamlessly. @iamareebjamal

**Comment by iamareebjamal** [comment]
There is a reason why we fail the build on indentation issues. It's by design, not an accident

**Comment by VamsiKrishnaCommits** [comment]
> There is a reason why we fail the build on indentation issues. It's by design, not an accident

@iamareebjamal sir, I now fixed it. As a beginner , I found the travis logs pretty vague. I feel the need to update or create a section dedicated to travis build .

**Comment by iamareebjamal** [comment]
```
* What went wrong:
Execution failed for task ':app:spotlessKotlinCheck'.
> The following files had format violations:
      app/src/main/java/org/fossasia/openevent/general/order/OrdersUnderUserFragment.kt
          @@ -76,9 +76,9 @@
           ········ordersUnderUserVM.connection
           ············.nonNull()
           ············.observe(viewLifecycleOwner,·Observer·{·isConnected·->
          -················if(!isConnected)
          +················if·(!isConnected)
           ····················showNoInternetScreen(true)
          -················else{
          +················else·{
           ················val·currentItems·=·ordersUnderUserVM.eventAndOrderPaged.value
           ················if·(currentItems·!=·null)·{
           ····················showNoInternetScreen(false)
          @@ -87,7 +87,7 @@
           ················}·else·{
           ····················ordersUnderUserVM.getOrdersAndEventsOfUser(showExpired·=·false,·fromDb·=·true)
           ················}
          -············}})
          +············}·})
           
           ········ordersUnderUserVM.numOfTickets
           ············.nonNull()
  Run './gradlew :app:spotlessApply' to fix these violations.
```

I don't think anyone in the world can call this vague, It literally mentions the solution right in the error message and you don't even have to manually change anything, just run a command and it will automatically fix it.

File: app/src/main/java/org/fossasia/openevent/general/order/OrdersUnderUserFragment.kt [hunk 1]
```diff
@@ -76,6 +76,10 @@ class OrdersUnderUserFragment : Fragment(), BottomIconDoubleClick {
         ordersUnderUserVM.connection
             .nonNull()
             .observe(viewLifecycleOwner, Observer { isConnected ->
+                if (!isConnected)
+                    showNoInternetScreen(true)
+
+                else {
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/order/OrdersUnderUserFragment.kt]: Still wrong indentation. I don't know why the build didn't fail. It should have

Comment (VamsiKrishnaCommits) [reply app/src/main/java/org/fossasia/openevent/general/order/OrdersUnderUserFragment.kt]: Do you want me to change that?

**Comment by VamsiKrishnaCommits** [comment]
> ```
> * What went wrong:
> Execution failed for task ':app:spotlessKotlinCheck'.
> > The following files had format violations:
>       app/src/main/java/org/fossasia/openevent/general/order/OrdersUnderUserFragment.kt
>           @@ -76,9 +76,9 @@
>            ········ordersUnderUserVM.connection
>            ············.nonNull()
>            ············.observe(viewLifecycleOwner,·Observer·{·isConnected·->
>           -················if(!isConnected)
>           +················if·(!isConnected)
>            ····················showNoInternetScreen(true)
>           -················else{
>           +················else·{
>            ················val·currentItems·=·ordersUnderUserVM.eventAndOrderPaged.value
>            ················if·(currentItems·!=·null)·{
>            ····················showNoInternetScreen(false)
>           @@ -87,7 +87,7 @@
>            ················}·else·{
>            ····················ordersUnderUserVM.getOrdersAndEventsOfUser(showExpired·=·false,·fromDb·=·true)
>            ················}
>           -············}})
>           +············}·})
>            
>            ········ordersUnderUserVM.numOfTickets
>            ············.nonNull()
>   Run './gradlew :app:spotlessApply' to fix these violations.
> ```
> 
> I don't think anyone in the world can call this vague, It literally mentions the solution right in the error message and you don't even have to manually change anything, just run a command and it will automatically fix it.

I will take back the word vague, I meant one should know how to read them . It became easy for me once I know what the logs meant.

File: app/src/main/java/org/fossasia/openevent/general/order/OrdersUnderUserFragment.kt (hunk 1 above)
Comment (iamareebjamal) [reply app/src/main/java/org/fossasia/openevent/general/order/OrdersUnderUserFragment.kt]: That's why I commented

Comment (VamsiKrishnaCommits) [reply app/src/main/java/org/fossasia/openevent/general/order/OrdersUnderUserFragment.kt]: Done sir.

Comment (iamareebjamal) [reply app/src/main/java/org/fossasia/openevent/general/order/OrdersUnderUserFragment.kt]: Not resolved

Comment (VamsiKrishnaCommits) [reply app/src/main/java/org/fossasia/openevent/general/order/OrdersUnderUserFragment.kt]: reason?  I removed the blank line which you apparently told was a wrong indentation practice.

Comment (iamareebjamal) [reply app/src/main/java/org/fossasia/openevent/general/order/OrdersUnderUserFragment.kt]: Not really. Blank line is not related to indentation, I have commented on the indentation if else block

Comment (VamsiKrishnaCommits) [reply app/src/main/java/org/fossasia/openevent/general/order/OrdersUnderUserFragment.kt]: sir, is it fine now?

Comment (VamsiKrishnaCommits) [reply app/src/main/java/org/fossasia/openevent/general/order/OrdersUnderUserFragment.kt]: sir, is it fine now?

File: app/src/main/java/org/fossasia/openevent/general/order/OrdersUnderUserFragment.kt [hunk 2]
```diff
@@ -76,6 +76,9 @@ class OrdersUnderUserFragment : Fragment(), BottomIconDoubleClick {
         ordersUnderUserVM.connection
             .nonNull()
             .observe(viewLifecycleOwner, Observer { isConnected ->
+                if (!isConnected) {
+                    showNoInternetScreen(true)
+                } else {
                 val currentItems = ordersUnderUserVM.eventAndOrderPaged.value
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/order/OrdersUnderUserFragment.kt]: Still wrong. Do you really not know what indentation should a block inside {} should have?

File: app/src/main/java/org/fossasia/openevent/general/order/OrdersUnderUserFragment.kt [hunk 3]
```diff
@@ -84,7 +87,7 @@ class OrdersUnderUserFragment : Fragment(), BottomIconDoubleClick {
                 } else {
                     ordersUnderUserVM.getOrdersAndEventsOfUser(showExpired = false, fromDb = true)
                 }
-            })
+            } })
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/order/OrdersUnderUserFragment.kt]: Wrong

**Review by iamareebjamal — APPROVED** [review]

File: app/src/main/java/org/fossasia/openevent/general/order/OrdersUnderUserFragment.kt (hunk 2 above)
Comment (VamsiKrishnaCommits) [reply app/src/main/java/org/fossasia/openevent/general/order/OrdersUnderUserFragment.kt]: I did not. Now I know sir.

**Pull request merged**